src/containerapp/azext_containerapp/_utils.py

```diff
@@ -436,10 +437,11 @@ def _get_oryx_run_image_version_key(tag_obj, framework):
 def parse_oryx_run_image(image_repository, tag) -> OryxRunImageTagProperty:
     # Example Oryx run image: mcr.microsoft.com/oryx/<platform>:<version>-<os>
     # Note: image_repository should NOT include the "mcr.microsoft.com/" prefix; it should ONLY be the repository
-    re_matches = re.findall(r"oryx\/([A-Za-z]*):([0-9.]*)-([A-Za-z-]*)", "{}:{}".format(image_repository, tag))
-    if len(re_matches) == 0:
+    re_matches = re.findall(r"(oryx|azure-buildpacks)\/([A-Za-z]*):([0-9.]*)-([A-Za-z-]*)",
+                            "{}:{}".format(image_repository, tag))
+    if len(re_matches) == 0 or len(re_matches[0]) < 4:
         return None
-    return dict(fullTag=tag, framework=re_matches[0][0], version=SemVer.parse(re_matches[0][1]), os=re_matches[0][2], architectures=None, support="lts")
+    return dict(fullTag=tag, framework=re_matches[0][1], version=SemVer.parse(re_matches[0][2]), os=re_matches[0][3], architectures=None, support="lts")
 
```

Comment:
using positional item is not good practice. Is it possible to use meaningful index to make sure correct item match?

Reply:
It's not possible, Python re module use index. Index indicates the position of the group in the regex.

Reply:
could you show me what the re_matches item looks like

Reply:
For regular expression-related code, we always recommend adding some examples or descriptions, otherwise no one except the author will be able to understand the meaning of these codes.